run.py

```diff
@@ -56,6 +57,19 @@ def py_shell():
     shell.interact()
 
 
+def python_runtime():
+    # extract the absolute script path to alter sys.path and specific dunder variables
+    script = pathlib.Path(sys.argv[2]).expanduser().absolute()
+    sys.path.insert(0, str(script.parent))
+    __file__ = str(script)
+
+    # update passed args so they don't start with "<binary> python"
+    sys.argv = sys.argv[2:]
+
+    with open(script) as f:
+        exec(f.read())
```

Comment:
```suggestion
def python_runtime():
    # extract the absolute script path to alter sys.path and specific dunder variables
    script = pathlib.Path(sys.argv[2]).expanduser().absolute()
    sys.path.insert(0, str(script.parent))

    # update passed args so they don't start with "<binary> python"
    sys.argv = sys.argv[2:]
    exec_locals = {"__name__": "__main__", "__file__": str(script), "__doc__": None}

    with open(script, encoding="utf-8") as rfh:
        exec(rfh.read(), exec_locals)
```

Also, do we really need to `sys.path.insert(0, str(script.parent))`?
The reason being that, we never had to to support this in tiamat-pip...
https://gitlab.com/saltstack/pop/tiamat-pip/-/blob/master/tiamatpip/cli.py#L332-397

Reply:
The `sys.path.insert` bothers me. Why it's needed when calling exec?

Reply:
@s0undt3ch @dwoz I've tried to mimic what Python executable does and it prepends the script basedir to the path
```bash
root@ip-172-31-45-201:/tmp# cat ~/test_path.py 
import sys
print(sys.path)
root@ip-172-31-45-201:/tmp# python3 ~/test_path.py 
['/root', '/usr/lib/python310.zip', '/usr/lib/python3.10', '/usr/lib/python3.10/lib-dynload', '/usr/local/lib/python3.10/dist-packages', '/usr/lib/python3/dist-packages']
```

And this is crucial if you want to use something besides single script - see example 5 in my comment, as this is the behaviour without the `sys.path.insert`:
```bash
root@ip-172-31-45-201:/tmp# cat ~/test_app/__init__.py 
import sub

print(sub.test())
root@ip-172-31-45-201:/tmp# cat ~/test_app/sub.py 
def test():
    return "ok"
root@ip-172-31-45-201:/tmp# python3 ~/test_app/__init__.py 
ok

root@ip-172-31-45-201:/tmp# /tmp/tmpmv5_gr0q/salt/dist/salt python ~/test_app/__init__.py 
Traceback (most recent call last):
  File "/root/.pyenv/versions/3.10.4/envs/tmpi0qtzmo4/bin/salt", line 73, in python_runtime
    exec(rfh.read(), exec_locals)
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'sub'
```

And the Tiamat package only sets own folders:
```
['/opt/saltstack/salt/pypath/lib/python3.10/site-packages', '/tmp/_MEIaNJxMa/base_library.zip', '/tmp/_MEIaNJxMa/lib-dynload', '/tmp/_MEIaNJxMa']
```
Which is what is really expected, just not in this case.

Sure, you can always work around this in the script itself with something like
```python
import sys
import os
sys.path.insert(0, os.path.dirname(os.path.realpath(__file__)))
```

so it works
```bash
root@ip-172-31-45-201:~# cat ~/test_app_2/__init__.py 
import sys
import os
sys.path.insert(0, os.path.dirname(os.path.realpath(__file__)))

import sub

print(sub.test())
root@ip-172-31-45-201:~# cat ~/test_app_2/sub.py 
def test():
    return "ok"
root@ip-172-31-45-201:~# cd /tmp/
root@ip-172-31-45-201:/tmp# /tmp/tmpmv5_gr0q/salt/dist/salt python ~/test_app_2/__init__.py 
ok
```

But my assumpton is that this should imitate what the standard python does - without the need of users to modify other code, just swap the executable.


@s0undt3ch Thanks, I've used your suggestion and added extra try/except to simulate the behaviour python executable does (as I mention the previous comment) - inspired by what you do in `tiamat-pip`

@Ch3LL thanks for the clarification, I can definitely do review of any test, or try to write them myself once this is merged